### PR TITLE
Fix actor shutdown to await merges

### LINKED ISSUE #4612: Make sure long merges are not cancelled on the Lambda indexer
When one invoke a lambda indexer to ingest a gz file, Quickwit will start an indexing and merge pipeline. As soon as the indexing is finished, we kill the merge pipeline. For long merges, this is going to be an issue as they will never finished.

### PR DESCRIPTION
**Pull request by rdettai**
### Description

Improve the actor shutdown on AWS Lambda runs so that merges go through. Fix #4612

### How was this PR tested?

I ran the Quickwit Lambda stack with the mock data generator for a few weeks with both:
- a 100k element per batch generator
- a 10 element per batch generator
Both ran without any error and all merges were properly  performed. 

This test also led to opening https://github.com/quickwit-oss/quickwit/issues/4734 as Cloudwatch logs costs where excessively high.

### CODE REVIEW

**Comment by guilload** [comment]
Can you revert the changes in code-generated files? I wonder how this is not breaking CI.

**Comment by rdettai** [comment]
Done!

**Review by trinity-1686a — COMMENTED** [review]
looks sensible, but i lack knowledge about this part of the codebase (and in particular cdk stuff) to be very confident in my assessment

File: distribution/lambda/cdk/stacks/services/quickwit_service.py [hunk 1]
```diff
@@ -12,8 +12,12 @@
 
 
 def extract_local_env() -> dict[str, str]:
-    """Extracts local environment variables that start with QW_LAMBDA_"""
-    return {k: os.environ[k] for k in os.environ.keys() if k.startswith("QW_LAMBDA_")}
+    """Extracts local environment variables QW_LAMBDA_* and QW_DISABLE_TELEMETRY"""
+    return {
+        k: os.environ[k]
+        for k in os.environ.keys()
+        if (k.startswith("QW_LAMBDA_") or k == "QW_DISABLE_TELEMETRY")
```
[the remark this answers is not in the document]
Comment (rdettai) [reply distribution/lambda/cdk/stacks/services/quickwit_service.py:19]: We currently don't support deletes in the Lambda deployment, so we might as well disable it. Does it really have a performance impact?

File: quickwit/quickwit-lambda/src/environment.rs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::env::var;
+
+use once_cell::sync::Lazy;
+
+pub static INDEX_ID: Lazy<String> =
+    Lazy::new(|| var("QW_LAMBDA_INDEX_ID").expect("QW_LAMBDA_INDEX_ID must be set"));
+
+pub static LOG_SPAN_BOUNDARIES: Lazy<bool> =
```
[the remark this answers is not in the document]
Comment (rdettai) [reply quickwit/quickwit-lambda/src/environment.rs:27]: yes, these are flags that I use to build the blog post visualizations and are not meant to be public.  A comment at least would be welcome, agreed

File: distribution/lambda/cdk/stacks/services/quickwit_service.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (trinity-1686a) [reply distribution/lambda/cdk/stacks/services/quickwit_service.py:19]: if you have enough indexes/splits, it has an impact (mostly in memory usage). For a reasonable deployment of lambdas, probably not actually

[the remark this answers is not in the document]
Comment (rdettai) [reply distribution/lambda/cdk/stacks/services/quickwit_service.py:19]: In general, the idea is to only expose configurations that are relevant for the Lambda deployment to make it easier to use.

**Review by trinity-1686a — APPROVED** [review]

**Pull request merged**